### ISSUE TITLE
Chapter 4: cannot assign 'torch.nn.modules.container.Sequential' as child module

### ISSUE THREAD

**Issue by Mzq12138** [falloutdurham/beginners-pytorch-deep-learning#64]
I just downgraded my torch and other packages to the correct version. Seems this doesn't work.

**Comment by MarcusFra** [comment]
Could you please elaborate a bit more on this problem? It might help if you show us the stacktrace and the versions you used. If you use torch < 1.8.x it might be due to wrong depencies to torchvision etc.. Have a look [here](https://pytorch.org/get-started/previous-versions/) for the correct versions to use 

I just ran chapter4.ipynb with the current versions torch 1.8.1 (+cu101) and torchvision 0.9.1 (+cu101) - and it's running without any problems. The same when I use torch==1.6.0 and torchvision==0.7.0.

**Comment by Mzq12138** [comment]
At first, I tried it with torch 1.8.1(+cpu) and torchvision 0.9.1(+cpu), I got this error:

I thought it might due to the torch version, then I tried downgrade torch&amp;torchvision, however I still have the same problem.

I googled but found nothing helpful.
I don't think it's the problem of cpu version torch.
help,plz.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "falloutdurham/beginners-pytorch-deep-learning"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年6月4日(星期五) 晚上8:54
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [falloutdurham/beginners-pytorch-deep-learning] Chapter 4: cannot assign 'torch.nn.modules.container.Sequential' as child module (#64)





 
Could you please elaborate a bit more on this problem? It might help if you show us the stacktrace and the versions you used. If you use torch < 1.8.x it might be due to wrong depencies to torchvision etc.. Have a look here for the correct versions to use
 
I just ran chapter4.ipynb with the current versions torch 1.8.1 (+cu101) and torchvision 0.9.1 (+cu101) - and it's running without any problems. The same when I use torch==1.6.0 and torchvision==0.7.0.

**Comment by MarcusFra** [comment]
Do you run the code on your local machine? Have you tried it on Colab?
At which line does the code break? 
```
transfer_model.fc = nn.Sequential(nn.Linear(transfer_model.fc.in_features,500),
                                  nn.ReLU(), nn.Dropout(), nn.Linear(500,2)) 
```
Here? Have you deleted a character (comma, bracket)  by incidence?

Is `cannot assign 'torch.nn.modules.container.Sequential' as child module` the only message you receive? That's a `TypeError`, but yet I cannot figure out where it stems from as I don't revceive the message.
Has the pretrained model been loaded correctly? (any error after excecution of `transfer_model = models.resnet50(pretrained=True)`?)

**Comment by Mzq12138** [comment]
I downloaded the original version of your tutorial from github, and I didn't get a syntax error. The pretrained model was loaded correctly. So I assume there is nothing wrong with the code.
I tried to reconstruct other layers of Resnet50 like this:

Then I get the same TypeError:

And then I tried to replace the fc layer with a single ReLU function:

It worked!

A single function to reconstruct the layer, OK!A sequential to replace the layer, NOT OK!
I assume this is some kinda unknown bug from pytorch, or it's just my machine's problem. lol






------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2021年6月5日(星期六) 下午3:29
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [falloutdurham/beginners-pytorch-deep-learning] Chapter 4: cannot assign &#39;torch.nn.modules.container.Sequential&#39; as child module (#64)





 
Do you run the code on your local machine? Have you tried it on Colab?
 At which line does the code break?
 transfer_model.fc = nn.Sequential(nn.Linear(transfer_model.fc.in_features,500),                                   nn.ReLU(), nn.Dropout(), nn.Linear(500,2))   
Here? Have you deleted a character (comma, bracket)  by incidence?
 
Is cannot assign 'torch.nn.modules.container.Sequential' as child module the only message you receive? That's a TypeError, but yet I cannot figure out where it stems from as I don't revceive the message.
 Has the pretrained model been loaded correctly? (any error after excecution of transfer_model = models.resnet50(pretrained=True)?)
 
&mdash;

**Comment by Mzq12138** [comment]
never mind, it worked somehow.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "falloutdurham/beginners-pytorch-deep-learning"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年6月5日(星期六) 下午3:29
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [falloutdurham/beginners-pytorch-deep-learning] Chapter 4: cannot assign 'torch.nn.modules.container.Sequential' as child module (#64)





 
Do you run the code on your local machine? Have you tried it on Colab?
 At which line does the code break?
 transfer_model.fc = nn.Sequential(nn.Linear(transfer_model.fc.in_features,500),                                   nn.ReLU(), nn.Dropout(), nn.Linear(500,2))   
Here? Have you deleted a character (comma, bracket)  by incidence?
 
Is cannot assign 'torch.nn.modules.container.Sequential' as child module the only message you receive? That's a TypeError, but yet I cannot figure out where it stems from as I don't revceive the message.
 Has the pretrained model been loaded correctly? (any error after excecution of transfer_model = models.resnet50(pretrained=True)?)

**Comment by MarcusFra** [comment]
OK.
I think I get what you mean, but unfortunately it seems that in your comment some explanations (after the respective colons) are missing/not printed.
